rosette/src/Ob.cc

```diff
@@ -58,7 +58,11 @@ extern int nontrivial_addr_to_pre_fixnum(Ob *);
 #endif
 
 extern "C" {
+#ifdef __GNUC__
+    char* getcwd (char*);
+#else
```

Comment:
I'm dubious of this use of __GNUC__. OSX has a getcwd and surely if you use clang on Linux, then you're going to have getcwd without having __GNUC__ defined?

Reply:
We're just trying to get it to build correctly on Ubuntu 16.04 with gcc.  We're not going try to support building on arbitrary *nix platforms.  Greg mixed fixed-size types with platform-dependent-size types; this is preliminary to those changes.

Reply:
It's not gcc, it's g++. And, Ubuntu has both g++ and clang. Really what you're doing is trying to get the thing running using a modern C++ compiler. The default C++ standard for g++ is C++14. C++11 is a reasonable choice, at this point. But, "gnu compiler" is that reasonable.

You can check the __cplusplus macro's value to determine the C++ standard at compile time:

http://www.stroustrup.com/C++11FAQ.html#11

Reply:
FWIW, I would go for C++14, it's a good standard.

Also, at this point I think getwd is deprecated and shouldn't be used. So, probably we can just elide that code and use getcwd without any macro checks at all?